### PR TITLE
feat: Clean up OpenCensus

### PR DESCRIPTION
**Pull request by TeodorSAP**
## Description

Changes proposed in this pull request (what was done and why):

- Remove OpenCensus-related code
- Add clean-up method for existing OpenCensus service

Changes refer to particular issues, PRs or documents:

- #739 

## Traceability
- [x] The PR is linked to a GitHub issue.
- [x] New features have a milestone set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] Adjusted the documentation if the change is user-facing.
- [ ] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

File: internal/reconciler/telemetry/reconciler.go [hunk 1]
```diff
@@ -105,6 +105,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, fmt.Errorf("failed to clean up old network policies: %w", err)
 	}
 
+	if err := r.cleanUpOpenCensus(ctx); err != nil {
```
Comment (shorim) [inline internal/reconciler/telemetry/reconciler.go:108]: maybe we can be more specific and rename it to `cleanUpOpenCensusService`

Comment (skhalash) [reply internal/reconciler/telemetry/reconciler.go:108]: Do we want to fail the reconciliation if cleanup fails? Maybe just log and swallow the error?

Comment (skhalash) [reply internal/reconciler/telemetry/reconciler.go:108]: @shorim What will you say?

File: README.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ For more information, see [Dynamic Logging Backend Configuration](https://github
 
 ### Traces
 
-The trace controller creates an [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) deployment and related Kubernetes objects from a `TracePipeline` custom resource. The collector is configured to receive traces using the OTLP and OpenCensus protocols, and forwards the received traces to a configurable OTLP backend.
+The trace controller creates an [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) deployment and related Kubernetes objects from a `TracePipeline` custom resource. The collector is configured to receive traces using the OTLP protocol, and forwards the received traces to a configurable OTLP backend.
```
[the remark this answers is not in the document]
Comment (shorim) [reply README.md:21]: If we are going to change if for the `Traces` section, then we should also change it in the `Metrics` section https://github.com/kyma-project/telemetry-manager/blob/1fea39c6788608919cb30758b0f04905657b9441/README.md?plain=1#L27 to keep it consistent

**Review by NHingerl — APPROVED** [review]

**Review by shorim — APPROVED** [review]

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (TeodorSAP) [reply README.md]: I will address this and see if we can change this in the whole repo, since this not a singular case.

**Pull request merged**